Google.Api.Generator/RoslynUtils/RoslynBuilder.cs

```diff
@@ -57,6 +64,35 @@ public static ClassDeclarationSyntax Class(Modifier modifier, Typ typ, TypeSynta
             return method;
         };
 
+        public static ParameterSyntax Parameter(TypeSyntax type, string name, ExpressionSyntax @default = null)
+        {
+            var param = SyntaxFactory.Parameter(Identifier(name)).WithType(type);
+            if (@default != null)
+            {
+                param = param.WithDefault(EqualsValueClause(@default));
+            }
+            return param;
+        }
+
+        public static PropertyDeclarationSyntax AutoProperty(Modifier modifiers, TypeSyntax type, string name, bool hasSetter = false)
+        {
+            var accessors = new List<AccessorDeclarationSyntax>
+            {
+                AccessorDeclaration(SyntaxKind.GetAccessorDeclaration).WithSemicolonToken(s_semicolonToken)
+            };
+            if (hasSetter)
+            {
+                accessors.Add(AccessorDeclaration(SyntaxKind.SetAccessorDeclaration).WithSemicolonToken(s_semicolonToken));
+            }
+            return PropertyDeclaration(type, name)
+                .AddModifiers(modifiers.ToSyntaxTokens())
+                .AddAccessorListAccessors(accessors.ToArray());
+        }
+
+        public static ExpressionSyntax Nameof(ParameterSyntax nameof) =>
+            InvocationExpression(IdentifierName("nameof"))
```

Comment:
There's a certain irony in requiring a string literal here, in code to avoid string literals.

Reply:
Indeed ;)
But at least this helper method means it's only here once :)